pkg/apis/wksprovider/controller/wksctl/machine_controller.go

```diff
@@ -843,7 +805,17 @@ func (a *MachineActuator) setNodeAnnotation(node *corev1.Node, key, value string
 	return nil
 }
 
-func (a *MachineActuator) setNodeLabel(node *corev1.Node, label, value string) error {
+func (a *MachineController) setNodeProviderIDIfNecessary(node *corev1.Node) error {
+	err := a.modifyNode(node, func(node *corev1.Node) {
+		node.Spec.ProviderID = "wks://" + node.Name
```

Comment:
We're mixing between wks and wksctl all the time. Maybe we should open an issue to decide on either?

Reply:
Yeah that one was a bit of a hurried kludge.  
I think it's only there to stop the capi-controller bitching; we don't use it, and I don't think we could rely on it because it would prevent wksctl from running at the same time as a cloud-provider.